pkg/splunk/controller/statefulset_test.go

```diff
@@ -52,3 +56,65 @@ func TestDefaultStatefulSetPodManager(t *testing.T) {
 	method := "DefaultStatefulSetPodManager.Update"
 	spltest.PodManagerTester(t, method, &mgr)
 }
+
+func TestSetStatefulSetOwnerRef(t *testing.T) {
+	cr := enterprisev1.Standalone{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "stack1",
+			Namespace: "test",
+		},
+	}
+
+	c := spltest.NewMockClient()
+	current := appsv1.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "splunk-test-monitoring-console",
+			Namespace: "test",
+		},
+	}
+	namespacedName := types.NamespacedName{Namespace: "test", Name: "splunk-test-monitoring-console"}
+
+	err := SetStatefulSetOwnerRef(c, &cr, namespacedName)
+	if err.Error() != "NotFound" {
+		t.Errorf("Couldn't detect missing owner reference %s", current.GetName())
+	}
+
+	// Create statefulset
+	err = splutil.CreateResource(c, &current)
+	if err != nil {
+		t.Errorf("Failed to create owner reference  %s", current.GetName())
+	}
+	//	current.OwnerReferences = nil
+
+	// Test existing owner reference
+	err = SetStatefulSetOwnerRef(c, &cr, namespacedName)
```

Comment:
Nit: For a little bit more easy coverage, we could call SetStatefulSetOwnerRef again to cover the existing OR scenario :)